source/extensions/health_checkers/mysql/mysql.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "envoy/config/health_checker/mysql/v2/mysql.pb.validate.h"
+#include "envoy/network/filter.h"
+
+#include "common/network/filter_impl.h"
+#include "common/upstream/health_checker_base_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HealthCheckers {
+namespace MySQLHealthChecker {
+
+/**
+ * MySQL health checker implementation. Sends PING and expects PONG.
+ */
+class MySQLHealthChecker : public Upstream::HealthCheckerImplBase {
+public:
+  MySQLHealthChecker(const Upstream::Cluster& cluster,
+                     const envoy::api::v2::core::HealthCheck& config,
+                     const envoy::config::health_checker::mysql::v2::MySQL& mysql_config,
+                     Event::Dispatcher& dispatcher, Runtime::Loader& runtime,
+                     Runtime::RandomGenerator& random);
+
+private:
+  typedef std::vector<uint8_t> Packet;
+
+  struct MySQLActiveHealthCheckSession;
+
+  struct MySQLSessionCallbacks : public Network::ConnectionCallbacks,
+                                 public Network::ReadFilterBaseImpl {
+    MySQLSessionCallbacks(MySQLActiveHealthCheckSession& parent) : parent_(parent) {}
+
+    // Network::ConnectionCallbacks
+    void onEvent(Network::ConnectionEvent event) override { parent_.onEvent(event); }
+    void onAboveWriteBufferHighWatermark() override {}
+    void onBelowWriteBufferLowWatermark() override {}
+
+    // Network::ReadFilter
+    Network::FilterStatus onData(Buffer::Instance& data, bool) override {
+      parent_.onData(data);
+      return Network::FilterStatus::StopIteration;
+    }
+
+    MySQLActiveHealthCheckSession& parent_;
+  };
+
+  struct MySQLActiveHealthCheckSession : public ActiveHealthCheckSession {
+    enum class Phase { Greeting, Reply, Over };
+
+    MySQLActiveHealthCheckSession(MySQLHealthChecker& parent, const Upstream::HostSharedPtr& host)
+        : ActiveHealthCheckSession(parent, host), parent_(parent) {}
+    ~MySQLActiveHealthCheckSession();
+
+    void onData(Buffer::Instance& data);
+    void onEvent(Network::ConnectionEvent event);
+
+    // ActiveHealthCheckSession
+    void onInterval() override;
+    void onTimeout() override;
+
+    bool parse_mysql_server_greeting_packet(Buffer::Instance& data);
```

Comment:
nit: can we make all these methods camelCase per normal Envoy style?

Reply:
of course. sorry about that, it's just muscle memory